aspnetcore/security/authorization/secure-data.md

```diff
@@ -14,305 +15,326 @@ uid: security/authorization/secure-data
+Update the `OnGetAsync` method so only approved contacts are shown to general users:
 
-[!code-csharp[Main](secure-data/samples/final/Controllers/ContactsController.cs?name=snippet_Edit)]
+[!code-csharp[Main](secure-data/samples/final2/Pages/Contacts/Index.cshtml.cs?name=snippet)]
 
-### Update the Delete method
+### Update the edit page model
 
-Update both `Delete` methods to use the authorization handler to verify the user owns the contact.
+Add an authorization handler to verify the user owns the contact. Because resource authorization is being validated, the `[Authorize]` attribute is not enough. The app doesn't have access to the resource when attributes are evaluated. Resource-based authorization must be imperative. Checks must be performed once the app has access to the resource, either by loading it in the page model or by loading it within the handler itself. You frequently access the resource by passing in the resource key.
 
-[!code-csharp[Main](secure-data/samples/final/Controllers/ContactsController.cs?name=snippet_Delete)]
+[!code-csharp[Main](secure-data/samples/final2/Pages/Contacts/Edit.cshtml.cs?name=snippet)]
 
-## Inject the authorization service into the views
+### Update Delete
 
-Currently the UI shows edit and delete links for data the user cannot modify. We'll fix that by applying the authorization handler to the views.
+Update the delete page model to use the authorization handler to verify the user has delete permission on the contact.
+
+[!code-csharp[Main](secure-data/samples/final2/Pages/Contacts/Delete.cshtml.cs?name=snippet)]
+
+## Inject the authorization service into the views
 
-Inject the authorization service in the *Views/_ViewImports.cshtml* file so it will be available to all views:
+Currently, the UI shows edit and delete links for data the user can't modify. The UI is fixed by applying the authorization handler to the views.
 
-[!code-html[Main](secure-data/samples/final/Views/_ViewImports.cshtml)]
+Inject the authorization service in the *Views/_ViewImports.cshtml* file so it's available to all views:
 
-Update the *Views/Contacts/Index.cshtml* Razor view to only display the edit and delete links for users who can edit/delete the contact.
+[!code-html[Main](secure-data/samples/final2/Pages/_ViewImports.cshtml?highlight=6-9)]
 
-Add `@using ContactManager.Authorization;`
+The preceding markup adds several `using` statements.
 
-Update the `Edit` and `Delete` links so they're only rendered for users with permission to edit and delete the contact.
+Update the `Edit` and `Delete` links in *Pages/Contacts/Index.cshtml* so they're only rendered for users with the appropriate permissions:
```

Comment:
The Edit and Delete text should be bolded instead of code fenced. I missed this one in my first pass.